src/Xrefcheck/Verify.hs

```diff
@@ -408,18 +410,17 @@ verifyReference
 
     checkRef mAnchor referredFile = verifying $ do
       checkReferredFileExists referredFile
-      case M.lookup referredFile repoInfo of
+      case join $ M.lookup referredFile repoInfo of
         Nothing -> pass  -- no support for such file, can do nothing
         Just referredFileInfo -> whenJust mAnchor $
           checkAnchor referredFile (_fiAnchors referredFileInfo)
 
     checkReferredFileExists file = do
-      let fileExists = readingSystem $ doesFileExist file
-      let dirExists = readingSystem $ doesDirectoryExist file
+      let fileExists =
+            any (equalFilePath $ normaliseWithNoTrailing file) $ M.keys repoInfo
+          isVirtual = matchesGlobPatterns root vcVirtualFiles file
 
-      let isVirtual = matchesGlobPatterns root vcVirtualFiles file
-
-      unless (fileExists || dirExists || isVirtual) $
```

Comment:
Hold on, we removed the `dirExists` check...

Won't this break links to directories? That reminds me, do we even have tests for links to directories?

/cc @Sorokin-Anton I think I forgot about these in #165 :sweat: I've reopened the issue, can you please have a look at [this comment](https://github.com/serokell/xrefcheck/issues/165#issuecomment-1276549285)?

Reply:
Oops, i forgot about directories. Thanks!